dali/pipeline/data/tensor_list.cc

```diff
@@ -791,11 +791,16 @@ void TensorList<Backend>::SetContiguity(BatchContiguity state) {
 
 
 template <typename Backend>
-void TensorList<Backend>::MakeContiguous(std::weak_ptr<void> owner) {
+void TensorList<Backend>::MakeContiguous(AccessOrder copy_order) {
   if (state_.IsContiguous()) {
     return;
   }
-  DALI_FAIL("Coalescing the buffer to Contiguous state is not yet implemented.");
+  auto contiguous_tmp = TensorList<Backend>(shape().num_samples());
+  contiguous_tmp.SetContiguity(BatchContiguity::Contiguous);
+  contiguous_tmp.SetupLike(*this);
+  contiguous_tmp.Copy(*this, order());
```

Comment:
Won't this already run on the `order()`. Both tmp and this already have it set.

Reply:
I don't think so:
```
  template <typename SrcBackend>
  void Copy(const TensorList<SrcBackend> &in_tv, AccessOrder order = {},
            bool use_copy_kernel = false);
```
so you need to provide an order, in this case I provide the order from `this`.

Reply:
Copy already does a three way merge between the dst order, source order and the order parameter I think.